code/workspaces/auth-service/src/dataaccess/userRolesRepository.js

```diff
@@ -17,6 +17,11 @@ async function getRoles(userId, userName) {
   return roles.toObject();
 }
 
+async function getOtherUserRoles(userId) {
```

Comment:
Isn't this essentially the same as `getRoles`?

Reply:
Not quite... `getRoles` is used for the active user, and if it's a first time user then they won't appear in Mongo, which is why there is the option to add a record, and which is why the userName is passed in (so that the userName appears in Mongo).
`getOtherUserRoles` is for a user guaranteed to be in Mongo, and so there's no option to create a user and userName isn't past in.